File: WeekendFlights.Domain/Entities/SeoPageContent.cs
namespace WeekendFlights.Domain.Entities;

public static class SeoPageTypes
{
    public const string WeekendFrom = "weekend-from";
    public const string DayTripsFrom = "day-trips-from";
    public const string WeekendOd = "weekend-od";
}

public class SeoFaqItem
{
    public string Q { get; set; } = "";
    public string A { get; set; } = "";
}

/// <summary>
/// Pre-generated unique copy for a programmatic SEO landing, in one locale.
/// </summary>
public class SeoPageContent
{
    public Guid Id { get; set; }

    /// <summary>weekend-from, day-trips-from, or weekend-od.</summary>
    public string PageType { get; set; } = null!;

    public string OriginCode { get; set; } = null!;

    /// <summary>Destination IATA for OD pages; empty for origin-only pages.</summary>
    public string DestinationCode { get; set; } = "";

    public string Locale { get; set; } = null!;

    public string Lead { get; set; } = "";

    public string Heading { get; set; } = "";

    public string MetaDescription { get; set; } = "";

    public List<string> Paragraphs { get; set; } = [];

    public List<SeoFaqItem> Faq { get; set; } = [];

    public string? SourceUrl { get; set; }

    public string? SourceTitle { get; set; }

    public DateTimeOffset GeneratedAt { get; set; }
}

File: flight-search/frontend/src/utils/seoPageContent.ts
export const SEO_PAGE_TYPES = {
  weekendFrom: 'weekend-from',
  dayTripsFrom: 'day-trips-from',
  weekendOd: 'weekend-od'
} as const;

export type SeoPageType = (typeof SEO_PAGE_TYPES)[keyof typeof SEO_PAGE_TYPES];

export interface SeoFaqItem {
  q: string;
  a: string;
}

export interface SeoPageContent {
  pageType: string;
  originCode: string;
  destinationCode: string;
  locale: string;
  lead: string;
  heading: string;
  metaDescription: string;
  paragraphs: string[];
  faq: SeoFaqItem[];
  sourceUrl?: string | null;
  sourceTitle?: string | null;
  generatedAt?: string;
}

export function seoContentKey(
  pageType: string,
  origin: string | null | undefined,
  destination: string | null | undefined,
  locale: string
): string {
  const type = pageType.trim().toLowerCase();
  const from = (origin ?? '').trim().toUpperCase();
  const to = (destination ?? '').trim().toUpperCase();
  const lang = locale.trim().toLowerCase();
  return to ? `${type}:${from}:${to}:${lang}` : `${type}:${from}:${lang}`;
}

export function isUsableSeoPageContent(content: SeoPageContent | null | undefined): content is SeoPageContent {
  return Boolean(content && Array.isArray(content.paragraphs) && content.paragraphs.length > 0);
}
